src/modules/queue/QueueComponent.jsx

```diff
@@ -21,79 +18,61 @@ class QueueContainer extends React.Component {
 	};
 
 	render() {
-		var queue = {};
-		if (this.state.queueOpen) {
-			queue.open = ' open';
-		} else {
-			queue.open = '';
+		if (!this.state.queueOpen) {
```

Comment:
Here's the biggest thing: Don't use `opacity: 0`, `display: none` and tricky CSS classes to display or not display components.

If you don't want to display a component, just don't create it. It's that easy.

Here, I say "If the queue is not open, display the queue button with the queue_img. Else display the whole list, and the queue button with the x_icon". 

Notice how in each branch I know whether it's open or not, so I don't have to do any tricky conditionals. I just render what I want for that case.